### ISSUE TITLE
Make sources clickable

### ISSUE THREAD

**Issue by JakeAdler** [JakeAdler/The-Scoop#18]
Not only draggable

**Comment by DearLorditsNate** [comment]
I started digging into this a little—thought it might be a straightforward implementation but actually seems pretty complex with the inclusion of the DND stuff. I was trying to harness the methods already in place for the rendering of content after the source is dropped into the NewsViewer. I tried this by passing down those methods from  as props to the SourceThumbnails so that they could be triggered with an onClick (instead of re-writing all of the "get content from the DB and render it to the page" methods) and essentially just say "fire off the same sequence of events as if this element were dropped." But triggered onClick. I did get that working with some test functions, but when I tried passing in any of the actual methods, they always threw errors.

I'm having a hard time parsing out what exactly each method does and it seems to be related to the DND stuff—maybe they are some methods you picked up from its documentation? Anyway, one of the primary issues is that you can only pass down one method to be fired on click. And it doesn't seem like all the DND methods funnel into one main method that calls back all the others—am I missing something there? Is there one? Maybe it could be refactored that way so that one method (as a prop) could be passed down to the Thumbnails which then triggers the whole sequence of events? That might be a way to achieve this.

Otherwise, I know you wanted to refactor the DND code anyway in another Issue. Maybe it'd just be easier to make it click-first and bring back in the DND stuff later?

**Comment by JakeAdler** [comment]
@DearLorditsNate As of now, the DND logic is kind of broken as it is, so as you said it will probably be easier to refactor the entire source --> viewer logic, handling clicks first and then adding in the DND code. Although there is probably a way to do this in the state that the source --> viewer code is in, I think we will probably thank ourselves later if we overhauled the patterns that we're currently using.

I think the best way to go about this is to rewrite the logic parts of most everything in `core-news`, since as of now in `NewsContainer.js` instead of any type of state management between the source menu and viewer, we wrap the two in the `<DndProvider>`  and almost all of the logic is relying on whatever type of state management React-DND is using under the hood. What are your thoughts @DearLorditsNate? Thanks for the help I appreciate the time you put into the comment.

**Comment by JakeAdler** [comment]
Note: Although I've been pushing straight into master for most tickets (since the project is so young), this would probably be a good point to start working in news branches especially given that the main UI logic is going to be rewritten.

**Comment by DearLorditsNate** [comment]
I agree! I think doing a little overhaul right now would be a good choice and that we would definitely thank ourselves later. I'm personally not familiar with the DND library and although I could dig in and figure some things out, I also agree that we shouldn't be relying on its hidden, under the hood, state management for such a crucial part of this application. We should be a little more hands-on with managing state, I think!

How users select sources and how those articles are populated on the page seems like a pretty central part of the application though so I don't want to barge in and start changing things too much. Do you want to take a stab at it first? If you're impartial though, I'm willing to give it a go. But I'd have a couple of initial questions:

1. Is initially selecting one of the sources when the user logs-in an essential part/intention of the application? As an alternative, maybe there's an aggregated feed that populates on page-load and then let the user filter down by source?
2. Do you always want the currently selected source to NOT appear as an option in the source menu?

Bonus question: How are the articles in the DB being populated? Will there be an option for the user to refresh the feed and scrape for new articles?

And yes, I am very pro-branch haha! Already been working on one as I've been playing around.
